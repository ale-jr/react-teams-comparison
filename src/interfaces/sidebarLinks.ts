import { IconDefinition } from "@fortawesome/fontawesome-svg-core"
export interface SidebarLink {
    icon?: IconDefinition
    title: string
    to: string
    links?: SidebarLink[]
}