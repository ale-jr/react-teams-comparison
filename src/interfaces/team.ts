export interface TeamValues {
  participation: number;
  eNPS: number;
  overallEngagement: number;
  recognition: number;
  feedback: number;
  relationshipWithPeers: number;
  relationshipWithManager: number;
  satisfaction: number;
  alignment: number;
  happiness: number;
  wellness: number;
  personalGrowth: number;
  ambassadorship: number;
}

export interface Team extends TeamValues {
  name: string;
}
