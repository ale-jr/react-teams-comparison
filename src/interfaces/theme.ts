export interface ColorVariants {
  darkest?: string;
  dark?: string;
  default: string;
  light: string;
  lightest?: string;
  contrast: string;
  "0"?: string;
  "50"?: string;
  "100"?: string;
  "200"?: string;
  "300"?: string;
  "400"?: string;
  "500"?: string;
  "600"?: string;
}

export type ColorVariantKeys = keyof ColorVariants;

export interface Sizing {
  xs: string;
  sm: string;
  md: string;
  lg: string;
  xl: string;
}

export type SizingKeys = keyof Sizing;

export interface Typography {
  size: Sizing;
  weight: {
    regular: string;
    semibold: string;
    bold: string;
  };
  font: {
    heading: string;
    body: string;
  };
}

export interface Colors {
  neutral: ColorVariants;
  primary: ColorVariants;
  success: ColorVariants;
  danger: ColorVariants;
}
export type ColorKeys = keyof Colors;

export interface Animation {
  duration: string;
  timingFunction: string;
  shorthand: string;
}

export interface Theme {
  colors: Colors;
  spacing: Sizing;
  typography: Typography;
  animation: Animation;
}
