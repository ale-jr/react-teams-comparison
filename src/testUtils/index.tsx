import { render, RenderOptions, RenderResult } from '@testing-library/react'
import ThemeProvider from 'providers/ThemeProvider'
import { ReactElement } from 'react'


const Providers: React.FC<{
    children: JSX.Element | JSX.Element[]
}> = ({ children }) => <ThemeProvider>{children}</ThemeProvider>


const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "queries">): RenderResult => render(ui, {
    wrapper: Providers,
    ...options
})

export * from '@testing-library/react'
export {customRender as render}