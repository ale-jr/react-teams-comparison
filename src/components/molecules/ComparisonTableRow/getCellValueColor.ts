import { Theme } from "interfaces/theme";

interface GetCellValueColorAttributes {
  theme: Theme;
  value?: number;
}

const LOWEST = 2.4;
const LOW = 4.9;
const HIGH = 7.4;
const HIGHEST = 10;

export const getCellValueColor = ({
  theme,
  value,
}: GetCellValueColorAttributes): string => {
  if (!value) return theme.colors.neutral["0"] || "";
  else if (value <= LOWEST) return theme.colors.danger.default;
  else if (value <= LOW) return theme.colors.danger.light;
  else if (value <= HIGH) return theme.colors.success.light;
  else if (value <= HIGHEST) return theme.colors.success.default;
  else return theme.colors.neutral["0"] || "";
};
