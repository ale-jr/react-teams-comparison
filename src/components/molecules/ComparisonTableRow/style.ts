import styled, { css } from "styled-components";
import { getCellValueColor } from "./getCellValueColor";
interface RowProps {
  type: "team" | "average" | "benchmark";
}

const benchmarkRowCss = css`
  td {
    background-color: ${({ theme }) => theme.colors.neutral["200"]};
    border: 1px solid ${({ theme }) => theme.colors.neutral["200"]};
  }

  td:first-child {
    font-weight: ${({ theme }) => theme.typography.weight.semibold};
  }
`;

const averageRowCss = css`
  td:first-child {
    font-weight: ${({ theme }) => theme.typography.weight.semibold};
  }
`;

export const Row = styled.tr<RowProps>`
  td {
    padding: ${({ theme }) => `${theme.spacing.xs} ${theme.spacing.sm}`};

    color: ${({ theme }) => theme.colors.neutral["600"]};
    font-weight: ${({ theme }) => theme.typography.weight.semibold};
    text-align: center;
  }

  td:first-child {
    border: 1px solid ${({ theme }) => theme.colors.neutral["200"]};

    font-weight: ${({ theme }) => theme.typography.weight.regular};
    text-align: start;
  }

  ${({ type }) => type === "average" && averageRowCss}
  ${({ type }) => type === "benchmark" && benchmarkRowCss}
`;

interface CellProps {
  value?: number;
}

export const Cell = styled.td<CellProps>`
  background-color: ${getCellValueColor};
  border: 1px solid ${getCellValueColor};
`;
