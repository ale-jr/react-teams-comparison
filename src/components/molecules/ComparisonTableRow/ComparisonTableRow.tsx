import { Team, TeamValues } from "interfaces/team";
import { Row, Cell } from './style'

interface ComparisonTableRow {
    team: Team
    type: 'team' | 'average' | 'benchmark'
}

const KEYS: (keyof TeamValues)[] = [
    'participation',
    'eNPS',
    'overallEngagement',
    'recognition',
    'feedback',
    'relationshipWithPeers',
    'relationshipWithManager',
    'satisfaction',
    'alignment',
    'happiness',
    'wellness',
    'personalGrowth',
    'ambassadorship'
]

const ComparisonTableRow: React.FC<ComparisonTableRow> = ({ team, type }) => {
    return <Row type={type}>
        <Cell>
            {team.name}
        </Cell>
        {
            KEYS.map(key => {
                const value = team[key]
                
                return <Cell value={type === "team" ? value : value} key={key}>
                    {value?.toFixed(1)}
                </Cell>
            })
        }
    </Row>
}

export default ComparisonTableRow