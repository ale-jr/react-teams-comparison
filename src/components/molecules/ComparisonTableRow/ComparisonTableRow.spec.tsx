import { Team } from 'interfaces/team'
import 'jest-styled-components'
import { render } from "testUtils"
import ComparisonTableRow from './index'


describe("ComparisonTableRow", () => {
    it("Should match snapshot", () => {
        const team: Team = {
            name: "Team A",
            alignment: 1,
            ambassadorship: 2,
            eNPS: 3,
            feedback: 4,
            happiness: 5,
            overallEngagement: 6,
            participation: 7,
            personalGrowth: 8,
            recognition: 9,
            relationshipWithManager: 10,
            relationshipWithPeers: 11,
            satisfaction: 12,
            wellness: 13
        }
        const teamComponent = render(<ComparisonTableRow team={team} type="team" />)
        expect(teamComponent.asFragment()).toMatchSnapshot()

        const averageComponent = render(<ComparisonTableRow team={team} type="average" />)
        expect(averageComponent).toMatchSnapshot()

        const benchmarkComponent = render(<ComparisonTableRow team={team} type="benchmark" />)
        expect(benchmarkComponent).toMatchSnapshot()
    })
})