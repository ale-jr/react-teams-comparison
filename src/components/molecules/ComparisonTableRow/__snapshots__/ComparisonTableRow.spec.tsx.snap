// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonTableRow Should match snapshot 1`] = `
<DocumentFragment>
  Team A7.03.06.09.04.011.010.012.01.05.013.08.02.0
</DocumentFragment>
`;

exports[`ComparisonTableRow Should match snapshot 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 td {
  padding: 0.375em 0.5em;
  color: #1C1E2A;
  font-weight: 600;
  text-align: center;
}

.c0 td:first-child {
  border: 1px solid #F1F1F5;
  font-weight: 400;
  text-align: start;
}

.c6 td {
  padding: 0.375em 0.5em;
  color: #1C1E2A;
  font-weight: 600;
  text-align: center;
}

.c6 td:first-child {
  border: 1px solid #F1F1F5;
  font-weight: 400;
  text-align: start;
}

.c6 td:first-child {
  font-weight: 600;
}

.c1 {
  background-color: #FFFFFF;
  border: 1px solid #FFFFFF;
}

.c2 {
  background-color: #DBF2E3;
  border: 1px solid #DBF2E3;
}

.c3 {
  background-color: #FFBEB8;
  border: 1px solid #FFBEB8;
}

.c4 {
  background-color: #9BE7B7;
  border: 1px solid #9BE7B7;
}

.c5 {
  background-color: #FF7777;
  border: 1px solid #FF7777;
}

<body>
    <div>
      <tr
        class="c0"
        type="team"
      >
        <td
          class="c1"
        >
          Team A
        </td>
        <td
          class="c2"
          value="7"
        >
          7.0
        </td>
        <td
          class="c3"
          value="3"
        >
          3.0
        </td>
        <td
          class="c2"
          value="6"
        >
          6.0
        </td>
        <td
          class="c4"
          value="9"
        >
          9.0
        </td>
        <td
          class="c3"
          value="4"
        >
          4.0
        </td>
        <td
          class="c1"
          value="11"
        >
          11.0
        </td>
        <td
          class="c4"
          value="10"
        >
          10.0
        </td>
        <td
          class="c1"
          value="12"
        >
          12.0
        </td>
        <td
          class="c5"
          value="1"
        >
          1.0
        </td>
        <td
          class="c2"
          value="5"
        >
          5.0
        </td>
        <td
          class="c1"
          value="13"
        >
          13.0
        </td>
        <td
          class="c4"
          value="8"
        >
          8.0
        </td>
        <td
          class="c5"
          value="2"
        >
          2.0
        </td>
      </tr>
    </div>
    <div>
      <tr
        class="c6"
        type="average"
      >
        <td
          class="c1"
        >
          Team A
        </td>
        <td
          class="c2"
          value="7"
        >
          7.0
        </td>
        <td
          class="c3"
          value="3"
        >
          3.0
        </td>
        <td
          class="c2"
          value="6"
        >
          6.0
        </td>
        <td
          class="c4"
          value="9"
        >
          9.0
        </td>
        <td
          class="c3"
          value="4"
        >
          4.0
        </td>
        <td
          class="c1"
          value="11"
        >
          11.0
        </td>
        <td
          class="c4"
          value="10"
        >
          10.0
        </td>
        <td
          class="c1"
          value="12"
        >
          12.0
        </td>
        <td
          class="c5"
          value="1"
        >
          1.0
        </td>
        <td
          class="c2"
          value="5"
        >
          5.0
        </td>
        <td
          class="c1"
          value="13"
        >
          13.0
        </td>
        <td
          class="c4"
          value="8"
        >
          8.0
        </td>
        <td
          class="c5"
          value="2"
        >
          2.0
        </td>
      </tr>
    </div>
  </body>,
  "container": .c0 td {
  padding: 0.375em 0.5em;
  color: #1C1E2A;
  font-weight: 600;
  text-align: center;
}

.c0 td:first-child {
  border: 1px solid #F1F1F5;
  font-weight: 400;
  text-align: start;
}

.c0 td:first-child {
  font-weight: 600;
}

.c1 {
  background-color: #FFFFFF;
  border: 1px solid #FFFFFF;
}

.c2 {
  background-color: #DBF2E3;
  border: 1px solid #DBF2E3;
}

.c3 {
  background-color: #FFBEB8;
  border: 1px solid #FFBEB8;
}

.c4 {
  background-color: #9BE7B7;
  border: 1px solid #9BE7B7;
}

.c5 {
  background-color: #FF7777;
  border: 1px solid #FF7777;
}

<div>
    <tr
      class="c0"
      type="average"
    >
      <td
        class="c1"
      >
        Team A
      </td>
      <td
        class="c2"
        value="7"
      >
        7.0
      </td>
      <td
        class="c3"
        value="3"
      >
        3.0
      </td>
      <td
        class="c2"
        value="6"
      >
        6.0
      </td>
      <td
        class="c4"
        value="9"
      >
        9.0
      </td>
      <td
        class="c3"
        value="4"
      >
        4.0
      </td>
      <td
        class="c1"
        value="11"
      >
        11.0
      </td>
      <td
        class="c4"
        value="10"
      >
        10.0
      </td>
      <td
        class="c1"
        value="12"
      >
        12.0
      </td>
      <td
        class="c5"
        value="1"
      >
        1.0
      </td>
      <td
        class="c2"
        value="5"
      >
        5.0
      </td>
      <td
        class="c1"
        value="13"
      >
        13.0
      </td>
      <td
        class="c4"
        value="8"
      >
        8.0
      </td>
      <td
        class="c5"
        value="2"
      >
        2.0
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ComparisonTableRow Should match snapshot 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 td {
  padding: 0.375em 0.5em;
  color: #1C1E2A;
  font-weight: 600;
  text-align: center;
}

.c0 td:first-child {
  border: 1px solid #F1F1F5;
  font-weight: 400;
  text-align: start;
}

.c6 td {
  padding: 0.375em 0.5em;
  color: #1C1E2A;
  font-weight: 600;
  text-align: center;
}

.c6 td:first-child {
  border: 1px solid #F1F1F5;
  font-weight: 400;
  text-align: start;
}

.c6 td:first-child {
  font-weight: 600;
}

.c7 td {
  padding: 0.375em 0.5em;
  color: #1C1E2A;
  font-weight: 600;
  text-align: center;
}

.c7 td:first-child {
  border: 1px solid #F1F1F5;
  font-weight: 400;
  text-align: start;
}

.c7 td {
  background-color: #F1F1F5;
  border: 1px solid #F1F1F5;
}

.c7 td:first-child {
  font-weight: 600;
}

.c1 {
  background-color: #FFFFFF;
  border: 1px solid #FFFFFF;
}

.c2 {
  background-color: #DBF2E3;
  border: 1px solid #DBF2E3;
}

.c3 {
  background-color: #FFBEB8;
  border: 1px solid #FFBEB8;
}

.c4 {
  background-color: #9BE7B7;
  border: 1px solid #9BE7B7;
}

.c5 {
  background-color: #FF7777;
  border: 1px solid #FF7777;
}

<body>
    <div>
      <tr
        class="c0"
        type="team"
      >
        <td
          class="c1"
        >
          Team A
        </td>
        <td
          class="c2"
          value="7"
        >
          7.0
        </td>
        <td
          class="c3"
          value="3"
        >
          3.0
        </td>
        <td
          class="c2"
          value="6"
        >
          6.0
        </td>
        <td
          class="c4"
          value="9"
        >
          9.0
        </td>
        <td
          class="c3"
          value="4"
        >
          4.0
        </td>
        <td
          class="c1"
          value="11"
        >
          11.0
        </td>
        <td
          class="c4"
          value="10"
        >
          10.0
        </td>
        <td
          class="c1"
          value="12"
        >
          12.0
        </td>
        <td
          class="c5"
          value="1"
        >
          1.0
        </td>
        <td
          class="c2"
          value="5"
        >
          5.0
        </td>
        <td
          class="c1"
          value="13"
        >
          13.0
        </td>
        <td
          class="c4"
          value="8"
        >
          8.0
        </td>
        <td
          class="c5"
          value="2"
        >
          2.0
        </td>
      </tr>
    </div>
    <div>
      <tr
        class="c6"
        type="average"
      >
        <td
          class="c1"
        >
          Team A
        </td>
        <td
          class="c2"
          value="7"
        >
          7.0
        </td>
        <td
          class="c3"
          value="3"
        >
          3.0
        </td>
        <td
          class="c2"
          value="6"
        >
          6.0
        </td>
        <td
          class="c4"
          value="9"
        >
          9.0
        </td>
        <td
          class="c3"
          value="4"
        >
          4.0
        </td>
        <td
          class="c1"
          value="11"
        >
          11.0
        </td>
        <td
          class="c4"
          value="10"
        >
          10.0
        </td>
        <td
          class="c1"
          value="12"
        >
          12.0
        </td>
        <td
          class="c5"
          value="1"
        >
          1.0
        </td>
        <td
          class="c2"
          value="5"
        >
          5.0
        </td>
        <td
          class="c1"
          value="13"
        >
          13.0
        </td>
        <td
          class="c4"
          value="8"
        >
          8.0
        </td>
        <td
          class="c5"
          value="2"
        >
          2.0
        </td>
      </tr>
    </div>
    <div>
      <tr
        class="c7"
        type="benchmark"
      >
        <td
          class="c1"
        >
          Team A
        </td>
        <td
          class="c2"
          value="7"
        >
          7.0
        </td>
        <td
          class="c3"
          value="3"
        >
          3.0
        </td>
        <td
          class="c2"
          value="6"
        >
          6.0
        </td>
        <td
          class="c4"
          value="9"
        >
          9.0
        </td>
        <td
          class="c3"
          value="4"
        >
          4.0
        </td>
        <td
          class="c1"
          value="11"
        >
          11.0
        </td>
        <td
          class="c4"
          value="10"
        >
          10.0
        </td>
        <td
          class="c1"
          value="12"
        >
          12.0
        </td>
        <td
          class="c5"
          value="1"
        >
          1.0
        </td>
        <td
          class="c2"
          value="5"
        >
          5.0
        </td>
        <td
          class="c1"
          value="13"
        >
          13.0
        </td>
        <td
          class="c4"
          value="8"
        >
          8.0
        </td>
        <td
          class="c5"
          value="2"
        >
          2.0
        </td>
      </tr>
    </div>
  </body>,
  "container": .c0 td {
  padding: 0.375em 0.5em;
  color: #1C1E2A;
  font-weight: 600;
  text-align: center;
}

.c0 td:first-child {
  border: 1px solid #F1F1F5;
  font-weight: 400;
  text-align: start;
}

.c0 td {
  background-color: #F1F1F5;
  border: 1px solid #F1F1F5;
}

.c0 td:first-child {
  font-weight: 600;
}

.c1 {
  background-color: #FFFFFF;
  border: 1px solid #FFFFFF;
}

.c2 {
  background-color: #DBF2E3;
  border: 1px solid #DBF2E3;
}

.c3 {
  background-color: #FFBEB8;
  border: 1px solid #FFBEB8;
}

.c4 {
  background-color: #9BE7B7;
  border: 1px solid #9BE7B7;
}

.c5 {
  background-color: #FF7777;
  border: 1px solid #FF7777;
}

<div>
    <tr
      class="c0"
      type="benchmark"
    >
      <td
        class="c1"
      >
        Team A
      </td>
      <td
        class="c2"
        value="7"
      >
        7.0
      </td>
      <td
        class="c3"
        value="3"
      >
        3.0
      </td>
      <td
        class="c2"
        value="6"
      >
        6.0
      </td>
      <td
        class="c4"
        value="9"
      >
        9.0
      </td>
      <td
        class="c3"
        value="4"
      >
        4.0
      </td>
      <td
        class="c1"
        value="11"
      >
        11.0
      </td>
      <td
        class="c4"
        value="10"
      >
        10.0
      </td>
      <td
        class="c1"
        value="12"
      >
        12.0
      </td>
      <td
        class="c5"
        value="1"
      >
        1.0
      </td>
      <td
        class="c2"
        value="5"
      >
        5.0
      </td>
      <td
        class="c1"
        value="13"
      >
        13.0
      </td>
      <td
        class="c4"
        value="8"
      >
        8.0
      </td>
      <td
        class="c5"
        value="2"
      >
        2.0
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
